header/Utility/TextSplitter.h

```diff
@@ -1,18 +1,17 @@
 #ifndef GPU_KARP_RABIN_TEXTSPLITTER_H
 #define GPU_KARP_RABIN_TEXTSPLITTER_H
 
-#include <cmath>
 #include <vector>
 
 class TextSplitter {
 
 private:
     static const int numOfThreads = 32;
 
-    size_t numOfWindows(size_t textLength, int patternLength);
-    std::pair<size_t, size_t> lenOfChains(size_t numOfWindows);
+    static size_t numOfWindows(size_t textLength, size_t patternLength);
+    static std::pair<size_t, size_t> lenOfChains(size_t numOfWindows);
 public:
-    std::vector<std::pair<size_t, size_t>> splitText(long long int textLength, int patternLength);
+    [[maybe_unused]] static std::vector<std::pair<size_t, size_t>> splitText(size_t textLength, size_t patternLength);
```

Comment:
Great use of `maube_unused` attribute, I haven't seen that never before, but it's great :100: